src/Html/HtmlParser.fs

```diff
@@ -355,9 +355,15 @@ module internal HtmlParser =
                     else TagEnd(name)
                 else Tag(false, name, x.GetAttributes())
 
+            // pre is the only default formatted tag, nested pres are not
+            // allowed in the spec.
+            if x.IsFormattedTag then
+                x.HasFormattedParent := not isEnd
+            else
+                x.HasFormattedParent := !x.HasFormattedParent || x.IsFormattedTag
```

Comment:
as an example if this is `mutable`-able, this could just be

`x.HasFormattedParent <- x.HasFormattedParent || x.IsFormattedTag`

The codebase is fairly old :)

Reply:
Hm, I just meant to be consistent, should I update it?